### PR TITLE
Adjust Kubernetes Tentacle startup to always refresh server comms address registration

### PR DESCRIPTION
**Pull request by scme0**
# Background

Recently we added the ability to define multiple `ServerCommsAddresses` to the Kubernetes Tentacle so that it could be configured to run with an Octopus Server HA Cluster setup.

The missing functionality was allowing users to adjust this configuration after installation.

# Results

With these changes, it's possible to adjust the configuration by doing a helm update:
```bash
helm upgrade --atomic \
--set agent.serverCommsAddresses="{http://host.docker.internal:10943/,http://host.docker.internal:10942/}" \
--reuse-values \
--create-namespace --namespace octopus-agent-scme \
scme \
/Users/scottmerchant/Documents/dev/helm-charts/charts/kubernetes-agent
```

[sc-75000]

### CODE REVIEW

File: source/Octopus.Tentacle/Configuration/ITentacleConfiguration.cs [hunk 1]
```diff
@@ -129,6 +129,8 @@ public interface IWritableTentacleConfiguration : ITentacleConfiguration
         /// </summary>
         void ResetTrustedOctopusServers();
 
+        bool SetTrustedOctopusServers(IEnumerable<OctopusServerConfiguration>? newServers);
```
Comment (kevjt) [inline source/Octopus.Tentacle/Configuration/ITentacleConfiguration.cs:132]: Is it worth adding a brief docstring here to say that this will overwrite the trusted servers with the new list?

File: source/Octopus.Tentacle/Commands/PollCommand.cs [hunk 2]
```diff
@@ -45,11 +46,14 @@ public class PollCommand : AbstractStandardCommand
             this.log = log;
             this.selector = selector;
 
-            api = AddOptionSet(new ApiEndpointOptions(Options));
+            // purposefully not adding this with AddOptionSet because
+            // we only want to validate it if reuseThumbprint is false.
+            api = new ApiEndpointOptions(Options);
```
Comment (kevjt) [inline source/Octopus.Tentacle/Commands/PollCommand.cs:51]: Understandable why this was done but it is hard to follow. If only validation was a separate method within `AbstractCommand` :(

File: source/Octopus.Tentacle/Commands/PollCommand.cs [hunk 3]
```diff
@@ -60,38 +64,60 @@ protected override void Start()
...
+
+            configuration.Value.AddOrUpdateTrustedOctopusServer(serverConfiguration);
+            VoteForRestart();
+
+            log.Info("Polling endpoint configured");
+        }
+
+        OctopusServerConfiguration CreateServerConfigurationFromFirstTrustedServer(Uri serverAddress)
+        {
+            var firstTrustedServer = configuration.Value.TrustedOctopusServers.First();
```
Comment (kevjt) [inline source/Octopus.Tentacle/Commands/PollCommand.cs:89]: Is this guaranteed to be what the user expects as the first server? i.e. it is the same as either the first server in the list of `server-comms-addresses`

Related question, what happens when the `server-comms-address` was also supplied along with the list of extra comms addresses. Does the "first" trusted server instance refer to the server from the singular server-comms-address because that is preferred in the configure-and-run.sh script?

**Review by kevjt — COMMENTED** [review]
Looks good, I just have a small question

File: source/Octopus.Tentacle/Commands/PollCommand.cs (hunk 2 above)
Comment (scme0) [reply source/Octopus.Tentacle/Commands/PollCommand.cs:51]: Yeah it's dumb because that's all that is done when an OptionSet is added via `AddOptionSet`.

File: source/Octopus.Tentacle/Configuration/ITentacleConfiguration.cs (hunk 1 above)
Comment (scme0) [reply source/Octopus.Tentacle/Configuration/ITentacleConfiguration.cs:132]: Good call, will do

File: source/Octopus.Tentacle/Commands/PollCommand.cs (hunk 3 above)
Comment (scme0) [reply source/Octopus.Tentacle/Commands/PollCommand.cs:89]: There doesn't seem to be any kind of sorting done to the TrustedServers so the first Server should be the one that is registered first which should be the one that's used when registering the agent with OctopusServer.

If `ServerCommsAddress` (singular) is provided it's used for the initial registration with Server (eg: when the tentacle registers itself as a deployment target with server). If it's not provided, the first entry in `ServerCommsAddresses` (plural) is used. The left over entries in `ServerCommsAddresses` are registered as additional addresses.

So if `ServerCommsAddress` is provided, the first server will be that one. If it's not the first server will be `ServerCommsAddresses[0]`.

Does that make sense?

Comment (kevjt) [reply source/Octopus.Tentacle/Commands/PollCommand.cs:89]: Yea that makes sense. Just wondering if it will be clear to the user 🤔

**Review by kevjt — APPROVED** [review]
lgtm

File: source/Octopus.Tentacle/Commands/PollCommand.cs (hunk 3 above)
Comment (scme0) [reply source/Octopus.Tentacle/Commands/PollCommand.cs:89]: I'm considering deprecating the singular form and we can remove it in the next major release of the helm chart 🤔
